### PR TITLE
adding a new feature from the practicals

### PR DESCRIPTION
**Pull request by fabiolrdiniz**
## Description
This PR adds a new feature from the practicals. This feature has been outlined in the Wednesday session.

## Definition of Done
A "bounds check" filter is added.

### Issue(s) addressed
None

## Dependencies
None

## Impact
None

### CODE REVIEW

**Review by gnosis84 — APPROVED** [review]
testing

**Review by LiaofanLin-NOAA — APPROVED** [review]
very good!

File: src/ufo/instantiateObsFilterFactory.h [hunk 1]
```diff
@@ -65,10 +66,12 @@ template<typename MODEL> void instantiateObsFilterFactory() {
            backgroundCheckRONBAMMaker("Background Check RONBAM");
   static oops::FilterMaker<MODEL, oops::ObsFilter<MODEL, ufo::TemporalThinning> >
            temporalThinningMaker("Temporal Thinning");
-  static oops::FilterMaker<MODEL, oops::ObsFilter<MODEL, ufo::PoissonDiskThinning> >
```
Comment (EricJames-NOAA) [inline src/ufo/instantiateObsFilterFactory.h:69]: Was it intentional to remove this PoissonDiskThinning?  Why was it removed?

Comment (fabiolrdiniz) [reply src/ufo/instantiateObsFilterFactory.h:69]: Hi @EricJames-NOAA. Thanks for checking it. I didn't remove it. I just moved it down below the code to be in alphabetical order. Please, see lines 71-72.

**Pull request merged**